PhpAmqpLib/Wire/IO/StreamIO.php

```diff
@@ -104,40 +104,46 @@ public function connect()
         $errstr = $errno = null;
 
         $remote = sprintf(
-            '%s://%s:%s', 
+            '%s://%s:%s',
             $this->protocol,
             $this->host,
             $this->port
         );
 
         set_error_handler(array($this, 'error_handler'));
-        
-        $this->sock = stream_socket_client(
-            $remote,
-            $errno,
-            $errstr,
-            $this->connection_timeout,
-            STREAM_CLIENT_CONNECT,
-            $this->context
-        );
-        
+
+        try {
+            $this->sock = stream_socket_client(
```

Comment:
It is going to throw an exception by the error handler that is being set by line 113.